### PR TITLE
fix(curriculum): Consolidated comments for JavaScript Algorithms and Data Structures challenges - part 3 of 4

### PR DESCRIPTION
**Pull request by RandellDawson**
Checklist:
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related to PR #38040

This PR is an attempt to consolidate/remove as many comments in the challenge seed code in the JavaScript Algorithms and Data Structures certification challenges, to be able to create a small translation lookup object for translating the English comments to other languages.

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.english.md
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.english.md]: The hint on line 19 no longer makes sense if we remove this. Either this could be moved into the description or the hint needs changing.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.english.md
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.english.md]: Again, the hint (on line 18, this time) stops making sense without this.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.english.md
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.english.md]: Same problem with the hint.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.english.md
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.english.md]: Hint again.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: I was not sure about this one.  I used the example in the code section as most of these earlier challenges had the user only making a slight change compared to the example in the code.  That is why I changed from the `sum` function that was there.  Let's see what @moT01 thinks.  I am flexible on this one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.english.md]: I tried to update the solution sections, but I may have missed some.  I think I would rather create a separate PR that removes all comments from the solutions.  What do you think?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.english.md]: Good catch. I will try to move the example into the description section.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.english.md]: Good catch. I will try to move the example into the description section.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.english.md]: Good catch. I will try to move the example into the description section.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.english.md]: Good catch. I will try to move the example into the description section.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.english.md
Comment (ojeytonwilliams) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.english.md]: That's fine, as long as we know what not to look for.  By the way, I'm leaving this conversation unresolved, because resolving it hides the conversation before the other person has a chance to read it.  I do think it's resolved, though!

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: I have reverted it back in a new commit.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: I have made this change in a new commit.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.english.md]: I added an example based on the challenge description.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.english.md]: I added an example based on the challenge description.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.english.md]: I moved the example to the challenge description.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.english.md]: I added an example based on the challenge description.

**Comment by RandellDawson** [comment]
@ojeytonwilliams I went ahead and fixed the merge conflicts on this one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/appending-variables-to-strings.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/appending-variables-to-strings.english.md]:
```suggestion
// Only change code below this line
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: This reads as -
>  This means we can take the return value of a function and assign it to a variable. numbers together, then:

`"numbers together, then:"` is probably left over from the previous commit undo...

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: What do you suggest?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/build-javascript-objects.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/build-javascript-objects.english.md]: This `ourDog` example doesn't seem to be placed back in the challenge description instead, unlike other similar challenges in the PR. Is it because this particular challenge already has 2 similar examples showing the usage (`cat` and `anotherObject`)?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/concatenating-strings-with-plus-operator.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/concatenating-strings-with-plus-operator.english.md]:
```suggestion
var myStr; // Only change this line
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/constructing-strings-with-variables.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/constructing-strings-with-variables.english.md]:
```suggestion
// ourStr is now "Hello, our name is freeCodeCamp, how are you?"
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-string-variables.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-string-variables.english.md]: Similar to the last comment, should the removed example be placed in the challenge description?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/delete-properties-from-a-javascript-object.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/delete-properties-from-a-javascript-object.english.md]:
```suggestion
*/ <- Is this line a typo?
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md]:
```suggestion
// Only change code above this line
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md]:
```suggestion
// Only change code above this line
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-string-variables.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-string-variables.english.md]: I did this in another PR.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md]: Done in another PR

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Minor fixes...

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/apply-functional-programming-to-convert-strings-to-url-slugs.english.md]: Done in another PR

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/concatenating-strings-with-plus-operator.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/concatenating-strings-with-plus-operator.english.md]: I was hoping to avoid having an `Only change this line`.  I will have to check if I added the `Only change code below this line` and `Only change code above this line` in a separate PR for this challenge.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/build-javascript-objects.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/build-javascript-objects.english.md]: Yes, I decided to remove this one, since the Description already had an example.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/appending-variables-to-strings.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/appending-variables-to-strings.english.md]: Fixed in a separate PR.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/delete-properties-from-a-javascript-object.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/delete-properties-from-a-javascript-object.english.md]: Yes it is a typo.  Can you make a suggestion to remove it and I will commit it?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (Manish-Giri) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: Unless I missed what you were going for, just removing "numbers together, then:" should fix it.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: This seems a little out of scope.  Is this related to an open PR?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by shivaylamba — CHANGES_REQUESTED** [review]
Merge conflicts

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: @Manish-Giri I actually fixed this in a separate commit based on a similar request by @ojeytonwilliams.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: @ojeytonwilliams The code above no longer appears on the `master` branch.  I think this is resolved.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: @ojeytonwilliams I have reverted this change in a separate commit.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md
Comment (ojeytonwilliams) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: Wouldn't this PR bring it back, though?

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.english.md]: @ojeytonwilliams Yes, you are correct.  I was looking at something else.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by moT01** [comment]
Hope you don't mind - I fixed the failing test that was accounting for one of the comments that was removed.

**Comment by RandellDawson** [comment]
@moT01 I don't mind at all.  I did not even notice it was failing.  Thanks.  

@Manish-Giri If you can review and approve, we can get this one merged.

**Comment by RandellDawson** [comment]
@Manish-Giri I think I have made all the changes you requested, so this PR just needs a final review and approval from you, so we can get it merged.  Once this one is merged, I will check one more time to make sure I have consolidated and documented all of the unique comments in the JavaScript Algorithms and Data Structures certification.

**Review by Manish-Giri — APPROVED** [review]
@RandellDawson Looks like I missed your last notification, my bad. Yes, this looks good now ✔️

**Comment by ojeytonwilliams** [comment]
@RandellDawson did you want check things before merging?  I'm happy for this to be merged regardless.

**Pull request merged**
